README.md

```diff
@@ -36,10 +36,10 @@ Getting started
 ---------------
 To try out the demo code in this repository, follow these steps:
 
-1. Install [objectbox-c](https://github.com/objectbox/objectbox-c) system-wide: `bash <(curl -s https://raw.githubusercontent.com/objectbox/objectbox-c/master/download.sh)` (answer Y when it asks about installing to /usr/lib).
-2. Back in this repository, run `pub get` to download all Dart dependencies.
-3. Finally run `dart test/test.dart` to start the demo script.
-   Note that, as fairly recent language features are used, the minimal required Dart version is 2.2.2.
+1. Install [objectbox-c](https://github.com/objectbox/objectbox-c) system-wide: `curl -s https://raw.githubusercontent.com/objectbox/objectbox-c/master/download.sh > /tmp/download.sh && bash /tmp/download.sh 0.7 && rm /tmp/download.sh` (answer Y when it asks about installing to /usr/lib).
+2. Back in this repository, run `pub get` in each of the following directories: `objectbox`, `objectbox_model_generator`, `objectbox_test`. If you just want to try out the tests, running it in `objectbox_test` is enough.
```

Comment:
The new branch doesn't follow conventions https://dart.dev/tools/pub/package-layout or rather it creates three packages.

What is the reason it must be split in such a way instead having a single package?
* lib instead of objectbox
* bin/binding-generator instead of objectbox_model_generator
* test instead of objectbox_test

Reply:
For this, I generally stuck to how Dart's [json_serializable](https://github.com/dart-lang/json_serializable) organizes its files. It uses source_gen as well (just like objectbox_model_generator), so it seemed like quite a good reference for some first orientation. Also, it subdivides the repository into three subprojects as well, each with its own `pubspec.yaml`: `example` (our `objectbox_test`), `json_annotation` (our `objectbox`) and `json_serializable` (our `objectbox_model_generator`). 

Unfortunately, source_gen and build_runner are rather poorly documented for the use case of generating source code for annotations, so it's hard to find good resources on how to better organize the directory structure. Apart from that, I also think that the model generator needs to have its own Dart package in order for [`build.yaml`](https://github.com/objectbox/objectbox-dart/blob/dev-2.5/objectbox_model_generator/build.yaml) to work properly.

Reply:
Core packages are not always a best source of organization since they use to have long history, before most conventions even emerged. 

The page I've linked (https://dart.dev/tools/pub/package-layout) is an official recommendation how a package should be structured and as such it would be strongly preferable to follow there are serious reasons we need to diverge. If you say building the generator is (I don't know enough of dart to help you there at the moment), then let's keep the generator as a separate project for now but at least leave the rest as a single one. What do you think?

Reply:
Great, I definitely agree; I'll also keep the generator as a separate package for now. Please check out my last commit, I reorganized the file structure similar to how you suggested.